### PR TITLE
Introduce CI, fix broken tests, and fix `brew install --HEAD mackerel-agent`

### PR DESCRIPTION
**Pull request by astj**
Mainly this p-r introduces GitHub Actions CI for brew Formulae, and fixes Formulae a little to make tests pass.

In fact I also have `brew audit` CI in my fork (see https://github.com/astj/homebrew-mackerel-agent/pull/2 ), but for now I'd like not to introduce it since it might be "too much".
If you think `brew audit`ing is useful (and welcome big fixes like https://github.com/astj/homebrew-mackerel-agent/pull/3/files due to RuboCop 🤖 rewriting single quotes to double quotes), please tell me and I'll update this p-r to include `brew audit`.

### CODE REVIEW

**Comment by astj** [comment]
Hmm workflow from fork does not run? Please see https://github.com/astj/homebrew-mackerel-agent/pull/2 to see how CIs running.

File: mackerel-agent.rb [hunk 1]
```diff
@@ -18,12 +18,9 @@ class MackerelAgent < Formula
 
   def install
     if build.head?
-      ENV['GOPATH'] = buildpath/'.go'
```
Comment (astj) [inline mackerel-agent.rb:21]: These are harmless, but useless so I removed them. (Maybe for GOPATH mode?)

File: mackerel-agent.rb [hunk 2]
```diff
@@ -18,12 +18,9 @@ class MackerelAgent < Formula
...
-      mkdir_p buildpath/'.go/src/github.com/mackerelio'
-      ln_s buildpath, buildpath/'.go/src/github.com/mackerelio/mackerel-agent'
       system 'make', 'build'
       bin.install 'build/mackerel-agent'
-      etc.install 'mackerel-agent.conf'
+      etc.install 'mackerel-agent.sample.conf' => 'mackerel-agent.conf'
```
Comment (astj) [inline mackerel-agent.rb:23]: `mackerel-agent.conf` itself does not exist in mackerel-agent repository. It's copied from `mackerel-agent.sample.conf` in `make crossbuild` task:

https://github.com/mackerelio/mackerel-agent/blob/68dc1cb7bb57612875894687c6a320fcbd5ef687/Makefile#L56

Comment (ne-sachirou) [reply mackerel-agent.rb:23]: So this was broken all the time. (☍﹏⁰)

File: mackerel-agent.rb [hunk 3]
```diff
@@ -32,7 +29,7 @@ def install
   end
 
   test do
-    system 'mackerel-agent', '-version'
+    system 'mackerel-agent', 'version'
```
Comment (ne-sachirou) [inline mackerel-agent.rb:32]: (☍﹏⁰)

File: mackerel-agent.rb (hunk 2 above)
Comment (astj) [reply mackerel-agent.rb:23]: Yes. maybe nobody tried `brew install --HEAD` for years……

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+          - mackerel-agent
+    steps:
+      # https://github.com/Homebrew/actions/tree/master/setup-homebrew
+      # This will clone this repo into $(brew --repo "$GITHUB_REPOSITORY")
+      - name: Set up Homebrew
+        id: set-up-homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+      - name: Cache Homebrew Bundler RubyGems
+        id: cache
+        uses: actions/cache@v1
```
Comment (ne-sachirou) [inline .github/workflows/ci.yml]: Please use actions/cache@v2 https://github.com/actions/cache/tree/v2

Comment (astj) [reply .github/workflows/ci.yml]: <del>https://github.com/mackerelio/homebrew-mackerel-agent/pull/31/commits/fad1de9f4290c43009d89339221e767113d06db2</del> => fixed commit message at https://github.com/mackerelio/homebrew-mackerel-agent/pull/31/commits/4b5a412781ec05fa184beeceaa9c848be90c1da3

**Comment by astj** [comment]
@ne-sachirou Hi. Sorry for late response. I fixed to use actions/cache@v2. Can you please review this again?

**Review by ne-sachirou — APPROVED** [review]
🙆‍♀️

**Pull request merged**